### PR TITLE
Infra: Combine manual and examples build

### LINKED ISSUE #338: Unify user manual and examples build
The idea would be to keep the example notebooks and user manual as separate content but unify the underlying build into a single sphinx site with shared poetry dependencies.

The whole python -> jupyter notebook workflow can also be retired.

### PR DESCRIPTION
**Pull request by CalMacCQ**
# Description

Quite a few changes to the repository here so I'll try to cover everything.

* Comibed manual and examples into a single sphinx build (single set of poerty dependencies and a single `conf.py file`. The content separation between manual and examples still exists but the build infrastructure is combined.
* notebooks are now executed and turned into html using the [myst-nb](https://myst-nb.readthedocs.io/en/latest/) library (replacing the [jupyter-book](https://jupyterbook.org/en/stable/intro.html) library). The manual pages are still built using the `jupyter_sphinx` extension for now.
* I've used the[ Quantinuum-sphinx theme](https://github.com/CQCL/quantinuum-sphinx) (built on top of [furo](https://github.com/pradyunsg/furo)) to align with the other parts of the tket docs.
* I've removed the python -> jupyter notebook conversion. 
* Note that there are some visual inconsistencies between the code cells for the manual and examples. These should be fixed by moving exclusively to `myst-nb` for the manual and examples (I've tested this and it works). I'll do it in a follow up PR so this one doesn't get more out of hand.
* I've removed the testing of the example notebooks mentioned in the `examples/ci-tested-notebooks.txt` file. All of the notebooks listed in the file are executed as part of the docs build now so there is no need to test execution twice.
* note that I've combined the examples and manual under "user-guide". It may be worth setting up the following redirects to avoid link breakage.

https://tket.quantinuum.com/examples/ -> https://tket.quantinuum.com/user-guide/
https://tket.quantinuum.com/user-manual/ -> https://tket.quantinuum.com/user-guide/

I'm not sure how to do this auto-forwarding so haven't done it yet in this PR

Screenshot 2024-07-30 at 14 58 18

* Finally as discussed it would be worth moving the rigetti and qiskit examples to the `pytket-pyquil` and `pytket-qiskit` repositories respectively.

https://github.com/CQCL/pytket-qiskit/issues/371
https://github.com/CQCL/pytket-pyquil/issues/95

# Related issues

Closes #338

### CODE REVIEW

File: docs/404.html [hunk 1]
```diff
@@ -1,17 +0,0 @@
-<!DOCTYPE html>
-<html>
-  <head>
-    <meta charset="utf-8"/>
-    <meta http-equiv="refresh" content="0; url=https://tket.quantinuum.com/user-manual/"/>
-    <script type="text/javascript">
-      window.location.href = "https://tket.quantinuum.com/user-manual/"
-    </script>
```
Comment (CalMacCQ) [inline docs/404.html:8]: Can this file be safely deleted now without breaking links?

**Review by aidanCQ — APPROVED** [review]

File: .github/workflows/check-docs-build.yml [hunk 2]
```diff
@@ -28,35 +28,28 @@ jobs:
             - '.github/**'
 
   check:
-    name: check manual
+    name: check docs build
     needs: changes
     if: github.event_name == 'schedule' || needs.changes.outputs.manual == 'true'
     runs-on: ubuntu-22.04
```
Comment (cqc-melf) [inline .github/workflows/check-docs-build.yml:34]: Do you know if there is a reason why we are not using `ubuntu-latest` ?

File: .github/workflows/check-docs-build.yml [hunk 3]
```diff
@@ -28,35 +28,28 @@ jobs:
...
         fetch-depth: '0'
     - run: git fetch --depth=1 origin +refs/tags/*:refs/tags/* +refs/heads/*:refs/remotes/origin/*
     - name: Set up Python 3.11
       uses: actions/setup-python@v5
       with:
         python-version: 3.11
-    - name: install python requirements for manual
+    - name: install python requirements for docs
       run: |
-        python -m pip install --upgrade pip
```
Comment (cqc-melf) [inline .github/workflows/check-docs-build.yml:46]: The removed packages are now all in poetry?

File: .gitmodules [hunk 4]
```diff
@@ -0,0 +1,4 @@
+[submodule "docs/quantinuum-sphinx"]
+	path = docs/quantinuum-sphinx
+	url = https://github.com/CQCL/quantinuum-sphinx.git
+	branch = dist
```
Comment (cqc-melf) [inline .gitmodules:4]: Is this a protected branch?

File: docs/.nojekyll
Comment (cqc-melf) [inline docs/.nojekyll:1]: What is this used for? Do we still need it?

File: docs/404.html
Comment (cqc-melf) [inline docs/404.html:1]: Why is this file deleted? I think it was doing the forwarding to tket.quantinuum.com? Is the forwarding now done somewhere else?

File: docs/_static/nav-config.js [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+    "navIconLinks": [
+        {
+            "title": "TKET Github",
+            "href": "https://github.com/CQCL/tket",
+            "pathMatch": "somewhere",
+            "iconImageURL": "_static/assets/github.svg",
+        },
+        {
+            "title": "TKET Slack Channel",
+            "href": "https://tketusers.slack.com/",
```
Comment (cqc-melf) [inline docs/_static/nav-config.js:30]: Is this a valid invite link?

File: docs/conf.py [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+templates_path = ["./quantinuum-sphinx/_templates/"]
+html_static_path = ["./quantinuum-sphinx/_static/", "_static/"]
+
+pytketdoc_base = "https://tket.quantinuum.com/api-docs/"
+
+intersphinx_mapping = {
+    "python": ("https://docs.python.org/3/", None),
+    "pytket": (pytketdoc_base, None),
+    "qiskit": ("https://docs.quantum.ibm.com/api/qiskit", None),
+    "pytket-qiskit": ("https://tket.quantinuum.com/extensions/pytket-qiskit/", None),
```
Comment (cqc-melf) [inline docs/conf.py:33]: Why do you have only two extensions in here?

File: docs/examples/algorithms_and_protocols/ansatz_sequence_example.ipynb
Comment (cqc-melf) [inline docs/examples/algorithms_and_protocols/ansatz_sequence_example.ipynb:1]: Should I review the notebooks, or are they just moved?

File: pyproject.toml [hunk 7]
```diff
@@ -1,2 +1,30 @@
...
+[tool.poetry]
+name = "pytket-docs"
+package-mode = false
+version = "0.1.0"
+description = ""
+authors = ["CalMacCQ <93673602+CalMacCQ@users.noreply.github.com>"]
+readme = "README.md"
+
+[tool.poetry.dependencies]
+python = "^3.10"
```
Comment (cqc-melf) [inline pyproject.toml:10]: Do we have dependabot set up to check for for updates?

**Review by cqc-melf — COMMENTED** [review]
A few questions. I think we also need to update the required checks for this PR to pass.

File: .github/workflows/check-docs-build.yml (hunk 2 above)
Comment (CalMacCQ) [reply .github/workflows/check-docs-build.yml:34]: I'm not sure. I can use the latest if you'd prefer?

File: docs/examples/algorithms_and_protocols/ansatz_sequence_example.ipynb
Comment (CalMacCQ) [reply docs/examples/algorithms_and_protocols/ansatz_sequence_example.ipynb:1]: Sorry diff is misleading. I have just moved the files. I haven't changed the content

File: docs/conf.py (hunk 6 above)
Comment (CalMacCQ) [reply docs/conf.py:33]: These are just the two most common  ones. I can add in some more.

File: docs/404.html
Comment (CalMacCQ) [reply docs/404.html:1]: I can add this back in

File: .gitmodules (hunk 4 above)
Comment (CalMacCQ) [reply .gitmodules:4]: Not sure

Is it protected @aidanCQ ?

File: docs/.nojekyll
Comment (CalMacCQ) [reply docs/.nojekyll:1]: I don't think this file is needed.

File: docs/404.html (hunk 1 above)
Comment (CalMacCQ) [reply docs/404.html:8]: Added this back in. There was some reformatting I think?

File: pyproject.toml (hunk 7 above)
Comment (CalMacCQ) [reply pyproject.toml:10]: Not currently I think. I can look into setting that up?

File: docs/_static/nav-config.js (hunk 5 above)
Comment (CalMacCQ) [reply docs/_static/nav-config.js:30]: I checked this and it seems to go to the correct page for people not in the workspace.

File: docs/404.html
Comment (CalMacCQ) [reply docs/404.html:1]: Added the `404.html` file back in

File: pyproject.toml (hunk 7 above)
Comment (CalMacCQ) [reply pyproject.toml:10]: Would you prefer to just check for pytket and extensions updates? Or the docs dependencies as well?

Comment (CalMacCQ) [reply pyproject.toml:10]: Maybe this dependabot config is already valid for `pyproject.toml`?

https://github.com/CQCL/pytket-docs/blob/main/.github/dependabot.yml

Comment (cqc-melf) [reply pyproject.toml:10]: Yes, that should probably cover everything.

**Review by cqc-melf — DISMISSED** [review]

**Review by cqc-melf — APPROVED** [review]
